packages/compiler-sfc/__tests__/__snapshots__/compileScriptPropsTransform.spec.ts.snap

```diff
@@ -80,9 +80,10 @@ exports[`sfc props transform default values w/ type declaration, prod mode 1`] =
 
 export default /*#__PURE__*/_defineComponent({
   props: {
-    foo: { default: 1 },
-    bar: { default: () => {} },
-    baz: null
+    foo: { type: Number, default: 1 },
+    bar: { type: Object, default: () => {} },
+    baz: { type: null },
+    bool: { type: Boolean }
```

Comment:
Since casting only applies to boolean props, we should only keep the type when it is `Boolean` to reduce generated code size.